core/src/test/java/io/parsingdata/metal/EqualityTest.java

```diff
@@ -196,24 +196,28 @@ public void parseGraph() {
 
     @Test
     public void stringRef() {
-        final Ref object = new NameRef("name");
-        assertFalse(object.equals(null));
+        final Ref<String> object = new NameRef("name");
+        assertNotEquals(null, object);
         assertNotEquals("name", object);
         assertEquals(object, new NameRef("name"));
+        assertNotEquals(object, new NameRef("name", "name"));
+        assertNotEquals(object, new NameRef("name", "otherName"));
         assertNotEquals(object, new NameRef("otherName"));
         assertNotEquals(object, new DefinitionRef(any("name")));
-        assertNotEquals(object, new NameRef("name", con(1)));
+        assertNotEquals(object, new NameRef( con(1), "name"));
```

Comment:
There is a rogue space before the `con(1)` ;)

Reply:
Removed! :)